addons/account/models/account_move.py

```diff
@@ -2513,6 +2514,31 @@ def _reverse_moves(self, default_values_list=None, cancel=False):
 
         return reverse_moves
 
+    def _action_invoice_ready_to_be_sent(self):
+        """ Hook allowing custom code when an invoice becomes ready to be sent by mail to the customer.
+        For example, when an EDI document must be sent to the government and be signed by it.
+        """
+        pass
+
+    def _is_ready_to_be_sent(self):
```

Comment:
Minor detail:
Shouldn't this be a computed fields?
It enables caching, the possibility to depend on it for other fields, and the possibility to display the value in the UI (for debugging purpose for instance)

(just resolve this if you don't agree)

Reply:
It could but I don't really want to fight with the upgrade team because I'm adding a new computed not-stored field...